key_action_map.cpp

```diff
@@ -417,22 +466,40 @@ void test_key_action_map()//!OCLINT tests can be many
     assert(kam.to_key(action_type::shoot_stun_rocket) == sf::Keyboard::E);
   }
 #endif // FIX_ISSUE_355
-
-//#define ISSUE_522
-#ifdef ISSUE_522
+  // 522. operator<< for key_action_map
   {
     const key_action_map kam = get_player_1_kam();
     std::stringstream s;
     s << kam;
   }
+  // 522. Can save and load a key_action_map
   {
     const key_action_map kam = get_player_1_kam();
     const std::string filename = "test.txt";
     save_to_file(kam, filename);
     const key_action_map map_again = load_kam(filename);
     assert(kam == map_again);
-    // TODO: delete temporary file
+    std::remove(filename.c_str());
+  }
+  #ifdef FIX_ISSUE_522
+  // 522. Can save and load a key_action_map
+  {
+    const key_action_map kam = get_player_2_kam();
+    const std::string filename = "test.txt";
+    save_to_file(kam, filename);
+    const key_action_map map_again = load_kam(filename);
+    assert(kam == map_again);
+    std::remove(filename.c_str());
+  }
+  // 522. Can save and load a key_action_map
```

Comment:
Does this title need to be repeated for every player?
```suggestion

```

Reply:
Unsure if it should. I enjoy every section of code to have its own heading and if two pieces of code do the same, the heading will be the same. But as it is more of a personal preference, I'd be wiling to remove it or give it another text.

Reply:
```suggestion
  // 522. Can save and load a key_action_map for player 3
```

Reply:
If it's all the same, I added an extra bit to be more specific ;)